lib/ansible/module_utils/basic.py

```diff
@@ -398,12 +401,12 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._verbosity = 0
         # May be used to set modifications to the environment for any
         # run_command invocation
-        self.run_command_environ_update = {}
-        self._clean = {}
+        self.run_command_environ_update: dict[str, str] = {}
+        self._clean: dict | str | None = {}
```

Comment:
I don't see any place in the module that treats this like a dict. The default is an abomination… This could probably be
```suggestion
        self._clean: str | None = None
```
instead.

Alternatively, we could allow an empty dict only:
```suggestion
        self._clean: dict[t.Never, t.Never] | str | None = {}
```

Reply:
I think the empty dict is the better choice for now. I'd prefer if this PR avoids behavior changes when not strictly necessary to pass type checking.

Reply:
So use the second suggestion, only allowing it to be empty on the typing level.

Reply:
`t.Never` is only supported for Python 3.11+. module utils typing must pass with Python versions from 3.9 to 3.13. So the second suggestion will not work.

Reply:
Older versions represent the same as `typing.NoReturn`. So this should work then:
```suggestion
        self._clean: dict[t.NoReturn, t.NoReturn] | str | None = {}
```

Alternatively, we could check if `typing-extensions` has it.